src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -222,10 +222,10 @@ def newest_timestamp(self) -> datetime | None:
 
         return self.time_bound_newest
 
-    def datetime_to_index(
+    def datetime_to_pos(
```

Comment:
I would rename `datetime_to_index()` to `datetime_to_internal_index()` instead, as when talking about positions in a container in Python the term `index` is usually used. In this case if we have like a *"physical"* and *"virtual"* indices, then I would just disambiguate by using `internal` (or `physical`). IMHO using `pos` / `position` just adds a different type of confusion, because is now the `position` the *virtual* index or the *physical* index?

I would also try to use a consistent nomenclature through the whole file, and maybe add a clarification in the class docs, saying what we mean when we talk about *internal index* and plain *index*.